### PR TITLE
Prog

### PR DESCRIPTION
**Pull request by ToriRori**

### CODE REVIEW

File: EGE/Prog/Lang.py [hunk 1]
```diff
@@ -10,27 +10,29 @@ class ops:
...
+        self.prio = {}
+        self.make_priorities()
 
-    def to_html(s: str):
+    def to_html(self, s: str):
         return html.escape(s)
 
     def op_fmt(self, op):
-        fmt = self.translate_op[op] or op
+        fmt = self.translate_op([op]) or op
```
Comment (tonichoff) [inline EGE/Prog/Lang.py]: I think here need remove [] in call translate_op

File: EGE/Prog/Lang.py [hunk 2]
```diff
@@ -10,27 +10,29 @@ class ops:
...
         return (
             fmt == '%' and '%%' or
             isinstance(fmt, str) and '%' in fmt and fmt or
             f"%s {fmt} %s")
 
     def un_op_fmt(self, op):
-        fmt = self.translate_un_op[op] or op
+        fmt = self.translate_un_op([op]) or op
```
Comment (tonichoff) [inline EGE/Prog/Lang.py]: And here too

**Review by tonichoff — COMMENTED** [review]
Suspicion on extra quad braces in some functions

File: EGE/Prog.py [hunk 3]
```diff
@@ -201,46 +199,51 @@ def _children(self): pass
...
+    def __init__(self, op, left=None, right=None):
+        super.__init__(op)
+        self.right = right
+        self.left = left
+
     def to_lang_fmt(self, lang):
-        lang.get_fmt('op_fmt', self.op
+        lang.get_fmt('op_fmt', self.op)
 
-    def children(self): return (self.left, self.right)
```
Comment (tonichoff) [inline EGE/Prog.py:237]: Why was it necessary to change tuple to list?

File: EGE/Prog.py [hunk 4]
```diff
@@ -271,43 +274,54 @@ def rotate_right(self):
...
 class Var(SynElement):
+    def __init__(self, name):
+        self.name = name
 
     def to_lang(self, lang):
-        returbn lang.get_fmt('var_fmt') % self.name
+        return lang.get_fmt('var_fmt') % self.name
 
     def run(self, env):
         v = env[self.name]
```
Comment (tonichoff) [inline EGE/Prog.py:322]: I'm suggesting to rename variable `v` to more informative

**Review by tonichoff — COMMENTED** [review]
Have one question and suggestion

File: EGE/Prog/Lang.py (hunk 1 above)
Comment (ToriRori) [reply EGE/Prog/Lang.py:28]: was not found usages of this functions

File: EGE/Prog.py (hunk 4 above)
Comment (ToriRori) [reply EGE/Prog.py:322]: Tried to save logic and names from perl. I suggest to leave it for refactoring later.

File: EGE/Prog/Lang.py (hunk 1 above)
Comment (tonichoff) [reply EGE/Prog/Lang.py]: With ToriRori did find usage of this functions. Need to write like this: `translate_op()[op]`

**Review by SergeyZinkovich — APPROVED** [review]

File: EGE/Prog.py (hunk 3 above)
Comment (ToriRori) [reply EGE/Prog.py:237]: returned back

**Review by tonichoff — APPROVED** [review]

**Pull request merged**